polkadot/rpc/src/lib.rs

```diff
@@ -99,6 +99,7 @@ pub fn create_full<C, P, SC, B>(
 ) -> Result<RpcExtension, Box<dyn std::error::Error + Send + Sync>>
 where
 	C: ProvideRuntimeApi<Block>
+		+ sc_client_api::BlockBackend<Block>
```

Comment:
nit: I think we don't need this if you use the `hash` method from `HeaderMetadata`